r/R/util.R

```diff
@@ -110,3 +110,12 @@ handle_embedded_nul_error <- function(e) {
   }
   stop(e)
 }
+
+handle_parquet_io_error <- function(e, format) {
+  msg <- conditionMessage(e)
+  if (grepl("Parquet magic bytes not found in footer", msg) && is.null(format)) {
+      e$message <- paste0(msg, "\nDid you mean to specify a 'format' other than the default (parquet)?")
+  }
+  stop(e)
```

Comment:
I think it's also a good place for: 

```
abort(c(
  msg, 
  i = "Did you mean to specify a 'format' other than the default (parquet)?"
))
```

Since this refers to `format`, would it be useful to add it as the argument list of `open_dataset()`, then I suppose we wouldn't need 

```r
args <- list2(...)
```

Reply:
That sounds like a good idea, but is it complicated by the fact that `open_dataset` can also take `Dataset` objects for the parameter `sources` and not just files?  If we were to add `format` as a parameter, if we keep the default value of `parquet`, does that feel weird if we're working with `Dataset` objects and not files? (Or could we just do it and mention in the docs?) And if we change `format` to be `NULL` by default, then that's quite a big breaking change, I guess.

I'm not sure what would be a good solution here or what the convention is.  What are your thoughts, @romainfrancois ?

Reply:
Agree that `abort(..., i = helpful_message)` is a good idea.

It probably is better to add `format` as an arg to `open_dataset()`, not because we need it there--it is just passed through--but it would help make it more explicit in the docs that it exists, what the default is, and what the supported values are (via match.arg). I don't think it's a problem that some kinds of inputs to open_dataset() wouldn't use the format argument: you can document that it is ignored unless `sources` is a string/character vector. And you wouldn't be making a breaking change or changing the defaults, you'd just be duplicating the `match.arg()` handling from DatasetFactory$create(). (This is the only downside I see, the duplication of that handling, but it's probably worth it on balance.)

Reply:
Makes sense; done now, just waiting for CI